File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/MedicalRecordsForm.ts
import { ServiceLookupEditor, DateEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface MedicalRecordsForm {
    DoctorId: ServiceLookupEditor;
    PatientId: ServiceLookupEditor;
    RecordDate: DateEditor;
    Notes: TextAreaEditor;
}

export class MedicalRecordsForm extends PrefixedContext {
    static readonly formKey = 'Hosbital.MedicalRecords';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MedicalRecordsForm.init)  {
            MedicalRecordsForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = DateEditor;
            var w2 = TextAreaEditor;

            initFormType(MedicalRecordsForm, [
                'DoctorId', w0,
                'PatientId', w0,
                'RecordDate', w1,
                'Notes', w2
            ]);
        }
    }
}

File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/MedicalRecords/MedicalRecordsForm.cs
using Serenity.ComponentModel;
using System;

namespace HosbitalSerenity.Hosbital.Forms;

[FormScript("Hosbital.MedicalRecords")]
[BasedOnRow(typeof(MedicalRecordsRow), CheckNames = true)]
public class MedicalRecordsForm
{
    public int DoctorId { get; set; }
    public int PatientId { get; set; }
    public DateTime RecordDate { get; set; }

    [TextAreaEditor(Rows = 5)]
    public string Notes { get; set; }
}